### PR TITLE
Jenkins: Reduce the number of stages per CI run

### PR DESCRIPTION
**Pull request by aserdean**
Building with `RelWithDebInfo` option does not give us a lot of value since
the Release version is compiled with optimization options and tested.

This will reduce the number of stages per run decreasing the job time and the
resources used.

Signed-off-by: Alin Gabriel Serdean <aserdean@ovn.org>

### CODE REVIEW

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -201,31 +201,23 @@ parallel "Check Developer Experience Ubuntu 16.04" :            { checkDevFlows(
...
          "ACC1604 clang-7 Debug" :                              { ACCTest('ACC-1604', 'clang-7', 'Debug') },
          "ACC1604 clang-7 Release" :                            { ACCTest('ACC-1604', 'clang-7', 'Release') },
-         "ACC1604 clang-7 RelWithDebInfo" :                     { ACCTest('ACC-1604', 'clang-7', 'RelWithDebinfo') },
          "ACC1604 gcc Debug" :                                  { ACCTest('ACC-1604', 'gcc', 'Debug') },
          "ACC1604 gcc Release" :                                { ACCTest('ACC-1604', 'gcc', 'Release') },
-         "ACC1604 gcc RelWithDebInfo" :                         { ACCTest('ACC-1604', 'gcc', 'RelWithDebInfo') },
          "ACC1604 Container RelWithDebInfo" :                   { ACCContainerTest('ACC-1604', '16.04') },
          "ACC1804 clang-7 Debug" :                              { ACCTest('ACC-1804', 'clang-7', 'Debug') },
          "ACC1804 clang-7 Release" :                            { ACCTest('ACC-1804', 'clang-7', 'Release') },
-         "ACC1804 clang-7 RelWithDebInfo" :                     { ACCTest('ACC-1804', 'clang-7', 'RelWithDebinfo') },
```
Comment (anitagov) [inline .jenkins/Jenkinsfile]: Good to retain ACC-1804 clang-7 RelWithDebinfo as well we will have one complete set exercised with the default configuration.

**Comment by johnkord** [comment]
Hey @aserdean do you know if there was a specific reason the "do not merge" label was added to this PR? I don't want to accidentally break anything if I merge this :)

**Comment by aserdean** [comment]
@johnkord: I added just so I could start the conversation around the topic :).

Removing the label now.

**Pull request merged**